packages/babel-plugin-proposal-destructuring-private/test/fixtures/ordering/.object-pattern-with-rest/exec.js

```diff
@@ -0,0 +1,18 @@
+var log = [];
+
+function push(x, y = x) { log.push(x); return y; }
+
+class C {
+  static #x;
+  static #y;
+  static {
+    var { [push(0)]: a = push(1), #x: {
+      [push(3)]: b = push(4),
+      #y: y = push(5),
+      [push(6)]: c = push(7)
+    } = push(2, C), [push(8)]: d = push(9), ...e } = C;
+  }
+}
+
+var nums = Array.from({ length: 10 }, (_, i) => i);
+expect(log).toEqual(nums);
```

Comment:
This test is disabled because `plugin-transform` evaluates `push(8)` immediately after `push(0)`, which should be fixed in a separate PR.